### ISSUE TITLE
Organize Redux types

### ISSUE THREAD

**Issue by tskarhed** [tskarhed/setlist#58]
The types are all in one file, I think it's mostly related to Redux. Some reorganization is needed, as it is getting quite complex. Any input is helpful! 

https://github.com/tskarhed/setlist/blob/master/src/state/types.ts

**Comment by PranjalAgni** [comment]
I think if we break the `reducers` and `action creators` in their own files then it would be more manageable in long term. We can also group the redux action types with its specific action creator

**Comment by tskarhed** [comment]
Sounds like a good idea! I am considering removing the specific actions, in favor of more general ones, such as `UPDATE_SONG`.  This is to easier sync with Firebase (https://github.com/tskarhed/setlist/pull/61)

**Comment by PranjalAgni** [comment]
Yes looking at number of action/reducers app has  currently, we need to first group the similar types of actions in their separate files and also take care while refactoring we do not miss something
